### ISSUE TITLE
Missing js files

### ISSUE THREAD

**Issue by Riku5543** [NiceneNerd/BCML#265]
Hey. I was debugging the white screen issue on my system and noticed that there is no bundle.js in the assets folder. In fact, there are no javascript files in the project. This issue occurs when downloading the latest released source as well as installing through pip. I think that's why people are getting white screens now.

**Comment by Riku5543** [comment]
It's worth noting that using the steps from bootstrap.sh builds the scripts and everything, but I'm still having trouble getting the frontend to display anything. I can go to the website from firefox though. (localhost:port)

**Comment by Riku5543** [comment]
I managed to fix the white screen by forcing the gui to cef instead of qt. It's important to know that this fix probably only applies to Linux. (Maybe mac idk) Also, you can't run configure_cef() because it's specific to Windows, but even without the setup, the cef view ends up working perfectly. I think this was the root cause of the white screen, and the missing js files may be an environment thing, I'm not sure. But this thread should help people with the white screen issue on Linux. (Or even windows if they need to use cef instead)

**Comment by GingerAvalanche** [comment]
CEF is now the default renderer on Windows, after the externally available Edge API broke rather spectacularly.

Please provide instructions on forcing the CEF renderer on Linux. PyQt5 has broken on Linux, so I would like to know this information as well, until such time as it gets fixed.

**Comment by Torphedo** [comment]
I was able to build from source and the latest version works fine for me on Windows...I can't tell if this issue is about Windows, Linux, or both

**Comment by GingerAvalanche** [comment]
The third comment explicitly states it's for Linux

**Comment by Torphedo** [comment]
I thought maybe they were having the issue on Windows and then tried on Linux or something...kind of a stupid assumption lmao

**Comment by Riku5543** [comment]
Okay sorry for the confusion. I did some digging with pywebview and it turns out that cef is not available on linux. (No surprise) I also tested `pip install bcml` again, and it turns out that the scripts are there normally. I'm not sure what happened before.

However, the qt renderer *really is* broken. Forcing it to gtk worked on my system after using `pip install gobject PyGObject`. I think setting gui to an invalid value defaults to gtk in my environment, so that's how I got it working before. Oh one last thing. Python 3.10.2 does not work correctly and I installed python 3.7 and used venv.

Sorry for all the confusion again.

TLDR:
White screen fix (works on Linux):
1. Use python 3.7 and create a venv.
2. `pip install bcml gobject PyGObject`
3. Open `__main__.py` in `venv/lib/python3.7/site-packages/bcml`
4. On line 108, set it to `gui = "gtk"`
5. Use `python -m bcml` to start bcml.

**Comment by NiceneNerd** [comment]
Bear in mind on some systems the GTK webview ends up working very oddly, with odd effects like double-input and other things, which is why it has never been used or recommended before.

**Comment by Riku5543** [comment]
That makes sense, but qt is showing nothing at the moment. I didn't have any issues with first-time setup or anything while using gtk. Maybe there could be a way to pick the renderer, either for all platforms or just for Linux. It would have to be before the webview is shown though. Maybe the app could read from a settings file that an external batch/bash script could modify before the first use.
